src/transformers/models/modernbert/configuration_modernbert.py

```diff
@@ -214,5 +214,10 @@ def __init__(
                 f'Invalid value for `classifier_pooling`, should be either "cls" or "mean", but is {self.classifier_pooling}.'
             )
 
+    def to_dict(self):
+        output = super().to_dict()
+        del output["reference_compile"]
```

Comment:
`pop()` with a default value is a little safer, as it won't crash if the value is missing or removed later
```suggestion
        output.pop("reference_compile", None)
```